swarmrl/agents/lymburn_model.py

```diff
@@ -0,0 +1,200 @@
+import h5py as hf
+import numpy as np
+
+from swarmrl.actions.actions import Action
+from swarmrl.agents.classical_agent import ClassicalAgent
+
+
+class Lymburn(ClassicalAgent):
+    def __init__(
+        self,
+        force_params: dict,
+        pred_movement: callable,
+        pred_params: np.array,
+        from_file: bool = False,
+        detection_radius_position_colls=np.inf,
+        detection_radius_position_pred=np.inf,
+        home_pos=np.array([500, 500, 0]),
+        agent_speed=10,
+        time_slice: float = 0.01,
+    ):
+        """
+        Parameters
+        ----------
+        force_params : dict
+            Dictionary containing the force parameters.
+                e.g.: force_params = {"K_a":K_a, "K_r":K_r,
+                  "K_h":K_h, "K_f":K_f, "K_p":K_p}
+        pred_movement : callable
+            Function that describes the movement of the predator
+                input: t, pos, velocity, home_pos, pred_params
+        pred_params : np. array (3,1)
+            Parameters for the predator movement function
+                If FromFile=True, force_params[0] is the path to
+                    the trajectory file
+        from_file: bool
+            If True, the predator follows a trajectory from a trajectory file
+        detection_radius_position_colls : float
+            Radius in which the colloids are detected
+        detection_radius_position_pred : float
+            Radius in which the predator is detected
+        home_pos : np. array
+            Position of the home
+        agent_speed : float
+            Speed of the colloids (used for the friction force)
+        time_slice : float
+        """
+        self.force_params = force_params
+        self.pred_movement = pred_movement
+        self.pred_params = pred_params
+        self.detection_radius_position_colls = detection_radius_position_colls
+        # implies r_align=r_repulsion
+        self.detection_radius_position_pred = detection_radius_position_pred
+        self.home_pos = home_pos
+        self.agent_speed = agent_speed
+
+        self.t = 0
+        self.time_slice = time_slice
+
+        self.from_file = from_file
+        if self.from_file:
```

Comment:
This particle should probably be somewhere else. Maybe you can make a `AgentFromTrajectory` agent which loads a particle from a trajectory and deploys it? Lymburn should just be the swarm itself. Not with a predator inside of it.